src/dex/fluid-dex/fluid-dex.ts

```diff
@@ -0,0 +1,881 @@
+import {
+  generalDecoder,
+  extractSuccessAndValue,
+  addressDecode,
+} from '../../lib/decoders';
+// @ts-ignore
+import { Tokens } from '../../../tests/constants-e2e';
+
+export class FluidDex extends SimpleExchange implements IDex<FluidDexData> {
+  readonly eventPools: { [id: string]: FluidDexEventPool } = {};
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly needWrapNative = false;
+  readonly isFeeOnTransferSupported = false;
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(FluidDexConfig);
+
+  logger: Logger;
+
+  pools: FluidDexPool[];
+
+  readonly FluidCommonAddresses: FluidDexCommonAddresses;
+
+  readonly iFluidDexPool: Interface;
+
+  protected adapters;
+
+  FEE_100_PERCENT = BigInt(1000000);
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+  ) {
+    super(dexHelper, dexKey);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.FluidCommonAddresses = new FluidDexCommonAddresses(
+      'FluidDex',
+      FluidDexConfig['FluidDex'][network].commonAddresses,
+      network,
+      dexHelper,
+      this.logger,
+    );
+    this.pools = FluidDexConfig['FluidDex'][network].pools;
+    this.adapters = Adapters[network] || {};
+    this.iFluidDexPool = new Interface(FluidDexPoolABI);
+  }
+
+  async fetchFluidDexPools(blockNumber: number): Promise<FluidDexPool[]> {
+    const poolsFromResolver =
+      await this.FluidCommonAddresses.getStateOrGenerate(blockNumber, false);
+
+    return poolsFromResolver.map(pool => ({
+      id: `FluidDex_${pool.address}`,
+      address: pool.address,
+      token0: pool.token0,
+      token1: pool.token1,
+    }));
+  }
+
+  async updatePoolAndEventPool(blockNumber: number) {
+    this.pools = await this.fetchFluidDexPools(blockNumber);
+    for (const pool of this.pools) {
+      if (!this.eventPools[pool.id]) {
+        this.eventPools[pool.id] = new FluidDexEventPool(
+          'FluidDex',
+          pool.address,
+          this.FluidCommonAddresses.commonAddresses,
+          this.network,
+          this.dexHelper,
+          this.logger,
+        );
+      }
+    }
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    Object.entries(this.eventPools).forEach(([id, eventPool]) => {
+      eventPool.getStateOrGenerate(blockNumber, false);
+    });
+    await this.updatePoolAndEventPool(blockNumber);
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  decodePools = (result: MultiResult<BytesLike> | BytesLike): Pool[] => {
```

Comment:
unused?

Reply:
Not sure if it is necessary to exist because was not useful in tests. 
But it is called by [src/dex/index.ts](https://github.com/bergben/paraswap-dex-lib/blob/master/src/dex/index.ts#L239), which may be useful in future.